linera-sdk/src/lib.rs

```diff
@@ -93,9 +95,25 @@ pub trait Contract: Sized {
     async fn initialize(
         &mut self,
         context: &OperationContext,
-        argument: &[u8],
+        argument: Self::InitializationArguments,
     ) -> Result<ExecutionResult, Self::Error>;
 
+    /// Initializes the application on the chain that created it.
+    ///
+    /// Performs serialization from `&[u8]` into [`Contract::InitializationArguments`] and delegates
+    /// to [`Contract::InitializationArguments`].
```

Comment:
Nit: I'd reword this documentation a bit, into something like:

```suggestion
    /// Deserializes the `argument` into [`Contract::InitializationArguments`] and calls
    /// [`Contract::initialize`].
```